tokio/src/runtime/blocking/pool.rs

```diff
@@ -267,6 +267,14 @@ impl Inner {
                 // Even if the condvar "timed out", if the pool is entering the
                 // shutdown phase, we want to perform the cleanup logic.
                 if !shared.shutdown && timeout_result.timed_out() {
+                    // Remove from list of worker threads to join
+                    let worker_thread_pos = shared
+                        .worker_threads
+                        .iter()
+                        .position(|x| x.thread().id() == thread::current().id())
```

Comment:
`thread::current()` includes an Arc clone (and drop). `thread::current()` should be called from outside of the loop and reused.

Reply:
After your suggestion, I don't even need `thread::current()` anymore. Also, it was not part of `loom` so...